services/core-web/common/actionCreators/permitActionCreator.ts

```diff
@@ -0,0 +1,500 @@
+        createRequestHeader()
+      )
+      .then((response) => {
+        notification.success({
+          message: "Successfully deleted permit amendment and all related documents.",
+          duration: 10,
+        });
+        dispatch(success(reducerTypes.DELETE_PERMIT_AMENDMENT));
+        return response;
+      })
+      .catch((err) => {
+        dispatch(error(reducerTypes.DELETE_PERMIT_AMENDMENT));
+        throw new Error(err);
+      })
+      .finally(() => dispatch(hideLoading()));
+  };
+
+export const fetchPermitConditions = (permitAmdendmentGuid) => (dispatch) => {
+  dispatch(request(reducerTypes.GET_PERMIT_CONDITIONS));
+  dispatch(showLoading());
+  return CustomAxios()
+    .get(
+      `${ENVIRONMENT.apiUrl}${API.PERMIT_CONDITIONS(null, null, permitAmdendmentGuid)}`,
+      createRequestHeader()
+    )
+    .then((response) => {
+      dispatch(success(reducerTypes.GET_PERMIT_CONDITIONS));
+      dispatch(permitActions.storePermitConditions(response.data));
+    })
+    .catch(() => dispatch(error(reducerTypes.GET_PERMIT_CONDITIONS)))
+    .finally(() => dispatch(hideLoading()));
+};
+
+export const createPermitCondition = (permitAmdendmentGuid, payload) => (dispatch) => {
+  dispatch(request(reducerTypes.CREATE_PERMIT_CONDITION));
+  dispatch(showLoading());
+  return CustomAxios()
+    .post(
+      `${ENVIRONMENT.apiUrl}${API.PERMIT_CONDITIONS(null, null, permitAmdendmentGuid)}`,
+      { permit_condition: payload },
+      createRequestHeader()
+    )
+    .then((response) => {
+      notification.success({
+        message: "Successfully created a new condition",
+        duration: 10,
+      });
+      dispatch(success(reducerTypes.CREATE_PERMIT_CONDITION));
+      return response;
+    })
+    .catch(() => dispatch(error(reducerTypes.CREATE_PERMIT_CONDITION)))
+    .finally(() => dispatch(hideLoading()));
+};
+
+export const deletePermitCondition = (permitAmdendmentGuid, permitConditionGuid) => (dispatch) => {
+  dispatch(request(reducerTypes.DELETE_PERMIT_CONDITION));
+  dispatch(showLoading("modal"));
+  return CustomAxios()
+    .delete(
+      `${ENVIRONMENT.apiUrl}${API.PERMIT_CONDITION(
+        null,
+        null,
+        permitAmdendmentGuid,
+        permitConditionGuid
+      )}`,
+      createRequestHeader()
+    )
+    .then((response) => {
+      notification.success({
+        message: "Successfully deleted permit condition.",
+        duration: 10,
+      });
+      dispatch(success(reducerTypes.DELETE_PERMIT_CONDITION));
+      return response;
+    })
+    .catch(() => dispatch(error(reducerTypes.DELETE_PERMIT_CONDITION)))
+    .finally(() => dispatch(hideLoading("modal")));
+};
+
+export const setEditingConditionFlag = (payload) => (dispatch) => {
+  dispatch(permitActions.storeEditingConditionFlag(payload));
+};
+
+export const updatePermitCondition =
+  (permitConditionGuid, permitAmdendmentGuid, payload) => (dispatch) => {
```

Comment:
same comment on types.

Reply:
ahh Just wanted to mention some of the functions in permitActionCreator weren't converted. I just converted the functions related to MinePermitInfo component.